scala-thoughts.md

```diff
@@ -0,0 +1,11 @@
+***What's Easy to do in Scala? What's hard?
+So far I think that I like pretty much everything about Scala. It really incorporates almost all of my favorite features from SML (a languge I initially hated but really really learned to love) while still having OOP and readable error messages (among other improvements). Obviously learning any language is hard and Scala is no exception, but despite my frustrations I really enjoyed my first spin. I think that Scala has a really really steep learning curve. Not an easy language to learn.
+
+***What are your favorite design choices?
+Love the arrows, love the pattern matching, love the emphasis on recursion. When I first learned recursion I was really scared of it but it is truly a beautiful and clean solution to many many problems. I also like how clean the if/else statements are. I also really like the way that it almost feels like java when you are doing OOP stuff but really feels like SML when pattern matching.
+
+***Least favorite design choices?
+I really dont like the way you access elements in ordered pairs. I'm not sure why the designers chose that and I think its not a very big deal. Just a bit counter intuitive. This isnt something that is bad at all, but there is not really any "right" way to do anything, just a variety of ways.
```

Comment:
Do you mean the ._1/._2 syntax?
It certainly would be nicer to just call MyPair(1) instead... I'm not sure why they chose to do it that way either.